### PR TITLE
Fix Flutter tools crash: `Bad state: Future already completed`

### PR DESCRIPTION
**Pull request by elliette**
Fixes https://github.com/flutter/flutter/issues/117676

See explanation https://github.com/flutter/flutter/issues/117676#issuecomment-1370353750.

### CODE REVIEW

**Comment by elliette** [comment]
Added `completer` back in, but no longer calling complete if the `completer` was already completed.

File: dwds/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,8 @@
   or not.
 - Pre-warm expression compiler cache to speed up Flutter Inspector loading.
 - Remove `ChromeProxyService.setExceptionPauseMode()`.
+- Don't complete an already completed `Completer` in `ChromeProxyService` to fix
+  Flutter tools crash: https://github.com/dart-lang/webdev/pull/1862
```
Comment (annagrin) [inline dwds/CHANGELOG.md:17]: I think we need to push this fix into a different branch of dwds, not master (a new branch created off the version used in flutter tools)

File: dwds/lib/src/services/chrome_proxy_service.dart [hunk 2]
```diff
@@ -287,7 +287,12 @@ class ChromeProxyService implements VmServiceInterface {
 
     unawaited(appConnection.onStart.then((_) async {
       await debugger.resumeFromStart();
-      _startedCompleter.complete();
+      if (!_startedCompleter.isCompleted) {
+        _startedCompleter.complete();
+      } else {
+        // See https://github.com/flutter/flutter/issues/117676:
+        _logger.warning('Expected state, debugging may not work as expected.');
```
Comment (annagrin) [inline dwds/lib/src/services/chrome_proxy_service.dart:294]: Unexpected state?

**Review by annagrin — COMMENTED** [review]
Thanks Elliott, left comments

File: dwds/lib/src/services/chrome_proxy_service.dart (hunk 2 above)
Comment (elliette) [reply dwds/lib/src/services/chrome_proxy_service.dart]: Oops, thank you!

File: dwds/lib/src/services/chrome_proxy_service.dart [hunk 3]
```diff
@@ -287,7 +283,6 @@ class ChromeProxyService implements VmServiceInterface {
 
     unawaited(appConnection.onStart.then((_) async {
       await debugger.resumeFromStart();
-      _startedCompleter.complete();
```
[the remark this answers is not in the document]
Comment (elliette) [reply dwds/lib/src/services/chrome_proxy_service.dart:290]: It isn't removed, it's just wrapped in an `if` statement

**Comment by elliette** [comment]
Closing in favor of https://github.com/dart-lang/webdev/pull/1867

**Pull request closed without merging**